### PR TITLE
Update code comments to improve readability in docs

### PR DESCRIPTION
**Pull request by soyrice**
Update code comments to improve readability and clarify the scope of the side input example.

R:
CC: @melap, @rezarokni

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java [hunk 1]
```diff
@@ -509,22 +509,22 @@ public static void fileProcessPattern() throws Exception {
...
-        // withCompression can be omitted - by default compression is detected from the filename.
+        // The withCompression method is optional. By default, the Beam SDK detects compression from the filename.
         .apply(FileIO.readMatches().withCompression(Compression.GZIP))
         .apply(
             ParDo.of(
                 new DoFn<FileIO.ReadableFile, String>() {
                   @ProcessElement
                   public void process(@Element FileIO.ReadableFile file) {
-                    // We now have access to the file and its metadata
+                    // We can now access to the file and its metadata.
```
Comment (lukecwik) [inline examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java]:
```suggestion
                    // We can now access the file and its metadata.
```

**Comment by soyrice** [comment]
Thanks lukecwik :) 

I'm having issues with the website precommit check - it's unable to access Docker and run ":website:testWebsite." (It throws "Process 'command 'docker' finished with non-zero exit value 1.") Do you have any advice for fixing this?

Since it's a minor change, I'll mark this PR ready for review now and then squash my commits once the website check passes.

**Comment by lukecwik** [comment]
The website precommit is failing due to a broken link:
```
*  linking to /documentation/patterns/side-input-patterns/#using-global-window-side-inputs-in-non-global-windows, but using-global-window-side-inputs-in-non-global-windows does not exist (line 413)
09:15:43      <a href="/documentation/patterns/side-input-patterns/#using-global-window-side-inputs-in-non-global-windows">Using global window side inputs in non-global windows</a>
```

**Comment by lukecwik** [comment]
Run Website PreCommit

File: website/src/documentation/patterns/side-input-patterns.md [hunk 2]
```diff
@@ -22,11 +22,11 @@ limitations under the License.
 
 The samples on this page show you common Beam side input patterns. A side input is an additional input that your `DoFn` can access each time it processes an element in the input `PCollection`. For more information, see the [programming guide section on side inputs]({{ site.baseurl }}/documentation/programming-guide/#side-inputs).
 
-## Using global window side inputs in non-global windows
+## Slowly updating global window side inputs
```
Comment (lukecwik) [inline website/src/documentation/patterns/side-input-patterns.md]: Changing this header here broke the link here: https://github.com/apache/beam/blame/39d695f040227a16766b83a36bf05c5f6d688000/website/src/documentation/patterns/overview.md#L30

You'll need to fix link to match your change to the header or revert your change to the header.

**Comment by lukecwik** [comment]
I opened #9024 with the fix to the broken link.

**Comment by lukecwik** [comment]
Merged as part of #9024

**Pull request closed without merging**